core/domain/wipeout_service.py

```diff
@@ -339,18 +347,160 @@ def _pseudonymize_models(activity_related_models, pseudonymized_user_id):
                 pseudonymized_user_id)
 
 
-def _delete_user_models(user_id):
-    """Delete the user models for the user with user_id.
+def _pseudonymize_feedback_models(pending_deletion_request):
+    """Pseudonymize the activity models for the user with user_id.
 
     Args:
-        user_id: str. The id of the user to be deleted.
+        pending_deletion_request: PendingDeletionRequest. The pending deletion
+            request object to be saved in the datastore.
     """
-    for model_class in models.Registry.get_storage_model_classes(
-            [models.NAMES.user]):
-        if (model_class.get_deletion_policy() not in
-                [base_models.DELETION_POLICY.KEEP,
-                 base_models.DELETION_POLICY.NOT_APPLICABLE]):
-            model_class.apply_deletion_policy(user_id)
+    user_id = pending_deletion_request.user_id
+
+    feedback_thread_model_class = feedback_models.GeneralFeedbackThreadModel
+    feedback_thread_models = feedback_thread_model_class.query(ndb.OR(
+        feedback_thread_model_class.original_author_id == user_id,
+        feedback_thread_model_class.last_nonempty_message_author_id == user_id
+    )).fetch()
+    feedback_ids = set([model.id for model in feedback_thread_models])
```

Comment:
Not sure what feedback_ids actually means here, since there are several "feedback-type" models. Could you use a more specific name that unambiguously defines which model type is being talked about here?

Reply:
The thing is that we use the same pid for all the models that share common feedback ID. See we can have a feedback (`GeneralFeedbackThreadModel`) that has multiple messages (`GeneralFeedbackMessageModel`) and also is a suggestion (`GeneralSuggestionModel`), the pseudonymous ID needs to stay the same on all these related models (for the user who requested the deletion), so after the pseudonymization the of feedback with one message and a suggestion this should be true `GeneralFeedbackThreadModel.original_author_id == GeneralFeedbackMessageModel.author_id == GeneralSuggestionModel.author_id`.

Reply:
This is really important context. Could you please add a detailed and clear comment to explain?

(In general, if a reviewer asks questions, you might want to do that proactively since if it's not clear to the reviewer then it's probably not clear to others who are looking at the codebase.)